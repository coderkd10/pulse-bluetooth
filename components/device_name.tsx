import { View, Text, StyleSheet } from "react-native";

export const DeviceName = ({name, id}: {name?: string, id: string}) => {
    return (
      <View>
        {name && <Text style={styles.deviceName}>{name}</Text>}
        <Text style={styles.deviceID}>{id}</Text>
      </View>
    );
}

const styles = StyleSheet.create({
    deviceName: {
      fontSize: 16,
    },
    deviceID: {
      color: '#666',
      fontWeight: 'bold',
    },
});
