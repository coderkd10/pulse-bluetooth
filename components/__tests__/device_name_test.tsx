import * as React from 'react';
import renderer from 'react-test-renderer';

import { DeviceName } from '../device_name';

it(`renders correctly`, () => {
  const tree = renderer.create(<DeviceName id="62:64:0D:5F:87:9C" name="JBL Flip" />).toJSON();

  expect(tree).toMatchSnapshot();
});
