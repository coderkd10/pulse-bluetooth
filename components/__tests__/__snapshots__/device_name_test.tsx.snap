// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    JBL Flip
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontWeight": "bold",
      }
    }
  >
    62:64:0D:5F:87:9C
  </Text>
</View>
`;
